src/components/providers/BackgroundTasksProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import BackgroundFetch from "react-native-background-fetch";
+
+type ContextState = {
+  isTrackingEnabled: boolean;
+  setTrackingEnabled?: (value: boolean) => void;
+};
+
+const initialState: ContextState = {
+  isTrackingEnabled: false,
+};
+
+const BackgroundTasksContext = createContext<ContextState>(initialState);
+
+const BackgroundTasksProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [isTrackingEnabled, setTrackingEnabled] = useState<boolean>(false);
+
+  useEffect(() => {
+    initBackgroundFetch();
+  }, []);
+
+  const initBackgroundFetch = async () => {
+    try {
+      const status: number = await BackgroundFetch.configure(
+        {
+          minimumFetchInterval: 15, // <-- minutes (15 is minimum allowed)
+          stopOnTerminate: false,
+          enableHeadless: true,
+          startOnBoot: true,
+          // Android options
+          forceAlarmManager: false, // <-- Set true to bypass JobScheduler.
+          requiredNetworkType: BackgroundFetch.NETWORK_TYPE_NONE, // Default
+          requiresCharging: false, // Default
+          requiresDeviceIdle: false, // Default
+          requiresBatteryNotLow: false, // Default
+          requiresStorageNotLow: false, // Default
+        },
+        async (taskId: string) => {
+          //TODO: add callback for the background task:
+          
+          console.log("[BackgroundFetch] taskId", taskId);
+
+          // Finish.
+          BackgroundFetch.finish(taskId);
+        },
+        (taskId: string) => {
+          console.log("[Fetch] TIMEOUT taskId:", taskId);
+          BackgroundFetch.finish(taskId);
+        }
+      );
+      console.log("BackgroundFetch configured successfully:", status);
+    } catch (error) {
+      console.error("BackgroundFetch configuration error:", error);
+    }
+
+    setTrackingEnabled(true);
```

Comment:
The state shouldn't change here, the way we can toggle this on is with the onClickToggleEnabled function

Reply:
Ok I removed that so that when it is configured it doesn't also start automatically. You have to toggle that with the button.

Reply:
if the flag is set to true and the task is not already running, it should start the task, but we can review this in a later PR

Reply:
Ok I'll make a note about that